mysql_protocol.cc

```diff
@@ -197,13 +197,50 @@ bool Protocol::Authenticate() {
   const char *authdata = user + user_len + 1;
   int authdata_len = Unpack(&authdata);  // length encoded
 
+  const char *clientplugin = reinterpret_cast<const char*>(authdata + authdata_len);
```

Comment:
Previously there was another "+ 1" here (it was "end + user_len + authdata_len + 2" and authdata is "end + user_len + 1"). Are you sure it's not needed?

Reply:
I think this is correct

Without the "+ 1":
```
W0319 08:12:24.692044  4559 mysql_protocol.cc:201] Client Authentication Plugin: caching_sha2_password
W0319 08:12:24.692073  4559 mysql_protocol.cc:204] Authenticating user: bbbbbbbb
I0319 08:12:24.692091  4559 mysql_protocol.cc:293] Sending AuthSwitchRequest for mysql_native_password
```

With the "+ 1":
```
W0319 08:13:05.399058  4878 mysql_protocol.cc:201] Client Authentication Plugin: aching_sha2_password
W0319 08:13:05.399081  4878 mysql_protocol.cc:204] Authenticating user: bbbbbbbb
I0319 08:13:05.399097  4878 mysql_protocol.cc:293] Sending AuthSwitchRequest for mysql_native_password
```
(here the first "c" of caching_sha2_password is skipped)